python/src/uagents/resolver.py

```diff
@@ -160,7 +198,9 @@ def __init__(self, max_endpoints: Optional[int] = None):
         self._max_endpoints = max_endpoints or DEFAULT_MAX_ENDPOINTS
         self._almanac_resolver = AlmanacResolver(max_endpoints=self._max_endpoints)
 
-    async def resolve(self, destination: str) -> Tuple[Optional[str], List[str]]:
+    async def resolve(
+        self, destination: str, test: bool
```

Comment:
Nitpicking a bit, but ideally `Resolver.resolve(...)` should take the same arguments for each resolver, as defined in the abstract `Resolver` class:
```
async def resolve(self, destination: str) -> Tuple[Optional[str], List[str]]:
```
And now that we have the concept of identifier, I think we could call it that in the functions:
```
async def resolve(self, identifier: str) -> Tuple[Optional[str], List[str]]:
```
Now it is clear that the `identifier` should have the all the information needed to resolve the endpoint, so it is the job of this `resolve` function to determine whether to go to the testnet or mainnet contracts.  Does that make sense?

Reply:
Yes, I was actually making that fix already 👍